src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+            size++;
+        } else {
+            Node<T> newNode = new Node<>(last, value, null);
+            last.next = newNode;
+            last = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            addFront(value);
+            return;
+        }
+        checkIndex(index);
```

Comment:
In my implementation, I can't do that, since my list starts with 0 elements. And moves the index to the next one. That is, in fact, for most checks for the presence of an element, it is OutOfBound, so it is in this case that the check should be done after these isolated cases. Or test's will break.